src/main/java/nextstep/config/AuthenticationPrincipalResolver.java

```diff
@@ -0,0 +1,49 @@
+package nextstep.config;
+
+import nextstep.exception.member.AuthTokenIsExpiredException;
+import nextstep.member.application.JwtTokenProvider;
+import nextstep.member.application.dto.AuthenticationPrincipal;
+import nextstep.utils.ObjectStringMapper;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Component
+public class AuthenticationPrincipalResolver implements HandlerMethodArgumentResolver {
+    private final JwtTokenProvider jwtTokenProvider;
+
+    public AuthenticationPrincipalResolver(JwtTokenProvider jwtTokenProvider) {
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+
+    @Override
+    public boolean supportsParameter(MethodParameter parameter) {
+        return parameter.getParameterAnnotation(AuthenticationPrincipal.class) != null &&
```

Comment:
```suggestion
        return parameter.hasParameterAnnotation(AuthenticationPrincipal.class) && ...
```

좀 더 적절한 메서드도 있습니다 ㅎㅎ